core/global_constants.cpp

```diff
@@ -388,90 +389,70 @@ void register_global_constants() {
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_0);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_1);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_2);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_3);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_4);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_5);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_6);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_7);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_8);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_9);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_10);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_11);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_12);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_13);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_14);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_15);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_MAX);
-
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_SONY_CIRCLE);
+	// Joypad buttons
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_INVALID_BUTTON);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_A);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_B);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_X);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_Y);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_BACK);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_GUIDE);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_START);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_LEFT_STICK);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_RIGHT_STICK);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_LEFT_SHOULDER);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_RIGHT_SHOULDER);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_DPAD_UP);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_DPAD_DOWN);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_DPAD_LEFT);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_DPAD_RIGHT);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_SDL_BUTTONS);
 	BIND_GLOBAL_ENUM_CONSTANT(JOY_SONY_X);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_SONY_CROSS);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_SONY_CIRCLE);
 	BIND_GLOBAL_ENUM_CONSTANT(JOY_SONY_SQUARE);
 	BIND_GLOBAL_ENUM_CONSTANT(JOY_SONY_TRIANGLE);
-
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_XBOX_B);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_SONY_SELECT);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_SONY_START);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_SONY_PS);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_SONY_L1);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_SONY_R1);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_SONY_L3);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_SONY_R3);
 	BIND_GLOBAL_ENUM_CONSTANT(JOY_XBOX_A);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_XBOX_B);
 	BIND_GLOBAL_ENUM_CONSTANT(JOY_XBOX_X);
 	BIND_GLOBAL_ENUM_CONSTANT(JOY_XBOX_Y);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_XBOX_BACK);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_XBOX_START);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_XBOX_HOME);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_XBOX_LS);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_XBOX_RS);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_XBOX_LB);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_XBOX_RB);
+	BIND_GLOBAL_ENUM_CONSTANT(JOY_BUTTON_MAX);
 
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_DS_A);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_DS_B);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_DS_X);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_DS_Y);
-
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_VR_GRIP);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_VR_PAD);
-	BIND_GLOBAL_ENUM_CONSTANT(JOY_VR_TRIGGER);
```

Comment:
So now the VR values are just gone? I use these, I would appreciate them staying in the engine...

Reply:
Do you have documentation on how the VR buttons and axes map onto the SDL buttons and axes? I removed them, because I didn't know how the mapped. The old values were limited to these three and they didn't make sense. Furthermore, not only has the order of the enums changed to align with the "official" SDL values, triggers are now axes and not buttons; so simply giving them the old values won't work either.

Reply:
I don't understand what you mean. My HTC Vive's triggers are dual-stage, same with my Steam controller. They are both buttons and axes.

If this PR removed support for such things, that's absolutely terrible and I think that part should be reverted. I don't see why adding half-axes requires removing dual-stage triggers.

Reply:
This PR implements the SDL mappings as SDL does. If your controller's triggers have two functions then my assumption is this would be reflected in their mappings. If not, then I'm missing something. It would be appreciated if you could test your controllers and provide feedback so things can be fixed if needed.